lib/IRGen/GenCall.cpp

```diff
@@ -2612,9 +2613,16 @@ class SyncCallEmission final : public CallEmission {
     if (fnConv.getNumDirectSILResults() == 1
         && (fnConv.getDirectSILResults().begin()->getConvention()
             == ResultConvention::Autoreleased)) {
-      if (IGF.IGM.Context.LangOpts.EnableObjCInterop)
-        result = emitObjCRetainAutoreleasedReturnValue(IGF, result);
-      else
+      if (IGF.IGM.Context.LangOpts.EnableObjCInterop) {
+        auto ty = fnConv.getSILResultType(IGF.IGM.getMaximalTypeExpansionContext());
+        auto *classTypeInfo = dyn_cast<ClassTypeInfo>(&IGF.IGM.getTypeInfo(ty));
+        if (classTypeInfo && classTypeInfo->getReferenceCounting() == ReferenceCounting::Custom) {
+          Explosion e(result);
+          classTypeInfo->strongCustomRetain(IGF, e, true);
```

Comment:
Since this only runs when ObjC interop is enabled, what would happen if we're running on e.g. Linux? Would this entire method not get executed at all?

Reply:
`swift_retain` is called if ObjC interop is disabled. See d0c35cf2d5c0a2a4c4c849e482f70e9677fdc517.

Reply:
Makes sense!